### ISSUE TITLE
Error when creating index having an icu_collation-filter with a custom rule

### ISSUE THREAD

**Issue by lindstromhenrik** [elastic/elasticsearch#27208]
When creating an index having an icu_collation-filter with a custom rule set the response is an error (invalid_path_exception).

**Elasticsearch version** (`bin/elasticsearch --version`): 
Version: 5.5.2, Build: b2f0c09/2017-08-14T12:33:14.154Z, JVM: 1.8.0_121

**Plugins installed**: [analysis-icu]

**JVM version** (`java -version`):
java version "1.8.0_131"
Java(TM) SE Runtime Environment (build 1.8.0_131-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.131-b11, mixed mode)

**OS version** (`uname -a` if on a Unix-like system):
Windows 10 Enterprise
Version: 1703

**Steps to reproduce**:

 1. Create index having a icu_collation-filter and a custom rule
```
PUT myindex
{
  "settings": {
    "analysis": {
      "analyzer": {
        "collation": {
          "tokenizer": "keyword",
          "type": "custom",
          "filter": [
            "mycollator"
          ]
        }
      },
      "filter": {
        "mycollator": {
          "type": "icu_collation",
          "rules": "& a < b"
        }
      }
    }
  }
}
```
Response:
```
{
  "error": {
    "root_cause": [
      {
        "type": "invalid_path_exception",
        "reason": "Illegal char <<> at index 4: & a < b"
      }
    ],
    "type": "invalid_path_exception",
    "reason": "Illegal char <<> at index 4: & a < b"
  },
  "status": 400
}
```

**Comment by cbuescher** [comment]
As stated in #27202 already, I cannot reproduce this using a clean 5.5.2 install on OS X, however since this error is observed on Windows, it might be possible to reproduce on that system.

**Comment by cbuescher** [comment]
@lindstromhenrik thanks, I was able to reproduce this now on Windows 7 with ES 5.5.2 myself. The way the rules are resolved seems to trigger slightly different exceptions on Windows than on other plattforms. I'll label this as a bug, thanks for your PR at #27202.

**Comment by jimczi** [comment]
I am not sure that we should fix this issue. The `icu_collation` filter is deprecated and should be removed in 7 (master) and maybe even in 6.x since the deprecation notice was added in 5.x.
The replacement for this is to use an ICU collation keyword field:
https://www.elastic.co/guide/en/elasticsearch/plugins/current/analysis-icu-collation-keyword-field.html
You can set special rules for this field but only as an inline string. Adding a path to a file is not allowed so the issue described here does not apply.

**Comment by cbuescher** [comment]
@jimczi thanks for pointing this out. Looking at how small the suggested fix in #27202 is, I think fixing it as a bug for the 6.x version should still be considered an option though. It doesn't seem to affect anything else seems to improves the situation at least. Would you comment on  #27202 if you have any concerns about the proposed fix?

**Comment by jimczi** [comment]
Yes it's unfortunate that we could not remove this in 6.0. We should at least make sure that new indices created in 6.1 cannot use this token filter. For older indices I agree that the fix is simple and can be added to 6.1 as well. I don't have any concern regarding the pr, just wanted to make sure that we're going to remove this filter as soon as we can.

**Comment by romseygeek** [comment]
cc @elastic/es-search-aggs

**Comment by romseygeek** [comment]
This was fixed by #27202 in 6.1
